hphp/hack/src/errors/nast_check_error.ml

```diff
@@ -445,7 +446,30 @@ let illegal_context pos name =
         Naming_special_names.Coeffects.contexts )
     []
 
-let case_fallthrough switch_pos case_pos =
+let case_fallthrough switch_pos case_pos next_pos =
+  let rec addspaces fallthrough_comment num_spaces =
+    if num_spaces > 0 then
+      addspaces (fallthrough_comment ^ " ") (num_spaces - 1)
+    else
+      fallthrough_comment
+  in
```

Comment:
This works, but appending to string in a loop is O(N^2). Some languages provide a way of doing this for efficiently, e.g. Java has `StringBuilder`.

For OCaml, you can use `String.make num_spaces ' '` ([docs here](https://ocaml.janestreet.com/ocaml-core/v0.12/doc/base/Base/String/index.html#val-make)).

Reply:
Ahh I was thinking there must be an easier way to do it!